### PR TITLE
Fix CreateMovieDetailsGroup() crash + validate parameters + small refactor

### LINKED ISSUE #1192: Crash in CreateMovieDetailsGroup()
## Software Versions

- Jellyfin Server Version:  N/A
- Roku Client Version:  v1.6.4

## Describe the bug
App crash found in `ShowScenes.brs` [line 356](https://github.com/jellyfin/jellyfin-roku/blob/v1.6.4/source/ShowScenes.brs#L356) via roku.com crash log

## Logs
```
'Dot' Operator attempted with invalid BrightScript Component or interface reference. (runtime error &hec) in pkg:/source/ShowScenes.brs(356) 
Backtrace: 
#1  Function createmoviedetailsgroup(movie As Dynamic) As Dynamic file/line: pkg:/source/ShowScenes.brs(356) 
#0  Function main(args As Dynamic) As Void file/line: pkg:/source/Main.brs(198) 
Local Variables: 
movie            roSGNode:MovieData refcnt=2 
global           Interface:ifGlobal m                roAssociativeArray refcnt=3 count:11 
group            roSGNode:MovieDetails refcnt=1 
moviemetadata    Invalid trailerdata      Invalid buttons          roSGNode:ButtonGroupHoriz refcnt=1 
b                roSGNode:Button refcnt=1 
extras           roSGNode:ExtrasRowList refcnt=2
```

## Screenshots
app-crashes

### PR DESCRIPTION
**Pull request by cewert**
I needed to do a small refactor to fix #1192 so I duplicated the small refactor to the other similar functions. The functions I didn't refactor I still added logic to verify the fucntion parameters before using them

## Changes
- Validate parameters for all functions in `ShowScenes.brs`
- Add parameter type to all functions
- Refactor some functions to grab meta data early and only proceed if meta data is not invalid (this was the cause of the crash)
- Rename a couple variables to be more explicit

## Issues
Fixes #1192

### CODE REVIEW

**Comment by cewert** [comment]
I tested this by loading my PR then cutting the internet to my roku. Instead of crashing when trying to view the movie details view, the app just sits there with a loading spinner and then fails silently and removes the spinner.

Ideally we should be logging any failed API calls and retrying them or attempting to reconnect to the server but this PR is just focused on preventing the crash.

File: source/VideoPlayer.brs [hunk 1]
```diff
@@ -1,4 +1,4 @@
-function VideoPlayer(id, mediaSourceId = invalid, audio_stream_idx = 1, subtitle_idx = -1, forceTranscoding = false, showIntro = true, allowResumeDialog = true)
+function VideoPlayer(id as string, mediaSourceId = invalid as dynamic, audio_stream_idx = 1 as integer, subtitle_idx = -1 as integer, forceTranscoding = false as boolean, showIntro = true as boolean, allowResumeDialog = true as boolean) as dynamic
```
Comment (1hitsong) [inline source/VideoPlayer.brs:1]: Makes sense to do these same changes to the new video player code so it works the same way.

`components\ItemGrid\LoadVideoContentTask.brs`

Comment (cewert) [reply source/VideoPlayer.brs:1]: fixed in latest commit

**Comment by cewert** [comment]
This will also fix another crash from the roku crash log that is similar to #1192 but in the `CreateSeasonDetailsGroup()` function

createseasondetailsgroup

**Review by 1hitsong — APPROVED** [review]

**Pull request merged**